plugins/mips/mips_arithmetic.ml

```diff
@@ -0,0 +1,234 @@
+  let im = signed imm ops.(1) in
+  RTL.[ rs := cpu.cia + im; ]
+
+(* ADDU rd, rs, rt
+ * Add Unsigned Word, MIPS32,
+ * Page 41 *)
+let addu cpu ops =
+  let rd = unsigned cpu.reg ops.(0) in
+  let rs = unsigned cpu.reg ops.(1) in
+  let rt = unsigned cpu.reg ops.(2) in
+  RTL.[ rd := rs + rt; ]
+
+(* CLO rd, rs
+ * Count Leading Ones in Word, MIPS32
+ * Page 136 *)
+(* TODO: check that RT = RD *)
+let clo cpu ops =
+  let rs = unsigned cpu.reg ops.(0) in
+  let rt = unsigned cpu.reg ops.(1) in
+  let rd = unsigned cpu.reg ops.(2) in
+  let xv = unsigned var word in
+  let cnt = unsigned var byte in
+  let has_no_zeroes = unsigned var bit in
+  let biti = unsigned var bit in
+  RTL.[
+    xv := low word rs;
+    cnt := zero;
+    has_no_zeroes := one;
+    foreach biti xv [
+      if_ (has_no_zeroes land (biti = one)) [
+        cnt := cnt + one;
+      ] [
+        has_no_zeroes := zero;
+      ]
+    ];
+    rd := cnt;
+  ]
+
+(* CLZ rd, rs
+ * Count Leading Zeroes in Word, MIPS32
+ * Page 137 *)
+(* TODO: check that RT = RD *)
+let clz cpu ops =
+  let rs = unsigned cpu.reg ops.(0) in
+  let rt = unsigned cpu.reg ops.(1) in
+  let rd = unsigned cpu.reg ops.(2) in
+  let xv = unsigned var word in
+  let cnt = unsigned var byte in
+  let has_no_ones = unsigned var bit in
+  let biti = unsigned var bit in
+  RTL.[
+    xv := low word rs;
+    cnt := zero;
+    has_no_ones := one;
+    foreach biti xv [
+      if_ (has_no_ones land (biti = zero)) [
+        cnt := cnt + one;
+      ] [
+        has_no_ones := zero;
+      ]
+    ];
+    rd := cnt;
+  ]
+
+(* LUi rt, imm
+ * Load Upper Immediate, MIPS32
+ * Page 279 *)
+let lui cpu ops =
+  let rt = unsigned cpu.reg ops.(0) in
+  let im = signed imm ops.(1) in
+  RTL.[
+    rt := im lsl unsigned const byte 16;
+  ]
+
+(* ALUIPC rs, imm
+ * Aligned Add Upper Immediate to PC, MIPS32 Release 6
+ * Page 45 *)
+let aluipc cpu ops =
+  let rs = signed cpu.reg ops.(0) in
+  let im = signed imm ops.(1) in
+  let x = unsigned var word in
+  let mask = unsigned var word in
+  RTL.[
+    mask := unsigned const word 0xFFFF;
+    x := cpu.cia + (im lsl unsigned const byte 16);
+    rs := (lnot mask) land x;
+  ]
+
+(* LSA rd, rs, rt, sa
+ * Load Scaled Address, MIPS32 Release 6
+ * Page 278 *)
+let lsa cpu ops =
+  let rd = unsigned cpu.reg ops.(0) in
+  let rs = unsigned cpu.reg ops.(1) in
+  let rt = unsigned cpu.reg ops.(2) in
+  let sa = unsigned imm ops.(3) in
+  RTL.[
+    rd := (rs lsl (sa + unsigned const byte 1)) + rt;
+  ]
+
+(* DLSA rd, rs, rt, sa
+ * Load Scaled Address, MIPS64 Release 6
+ * Page 278 *)
+let dlsa cpu ops =
+  let rd = unsigned cpu.reg ops.(0) in
+  let rs = unsigned cpu.reg ops.(1) in
+  let rt = unsigned cpu.reg ops.(2) in
+  let sa = unsigned imm ops.(3) in
+  RTL.[
+    rd := (rs lsl (sa + unsigned const byte 1)) + rt;
+  ]
+
+(* MOVE rd, rs  - WTF nonexistent!? *)
+
+(* SEB rd, rt
+ * Sign-Extend Byte, MIPS32 Release 2
+ * Page 426 *)
+let seb cpu ops =
```

Comment:
it's a not correct implementation. If I understood it right, we should take a least significant byte from rt, sign-extend it, and store in rs. 
So I would do it something like following:
```
let seb cpu ops =
    let rt = unsigned cpu.reg ops.(0) in
    let rs = signed cpu.reg ops.(1) in
    RTL.[
        rs := last byte rt;
    ]
```